homeassistant/components/media_player/cast.py

```diff
@@ -68,12 +68,17 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
 
     casts = []
 
+    all_chromecasts = pychromecast.get_chromecasts()
```

Comment:
do you think this is an important fix? `discover_chromecasts` is only called in one of the `if` cases, and in most scenarios there won't be more than 1 anyways. Even when there are 3 like in my case, with 8 groups overall, I didn't notice any major slowdown. All of this is to say I wouldn't recommend a special handling just for the case where we call `discover_chromecasts` as it would result in harder to maintain code.